bootstrap/README.md

```diff
@@ -77,3 +84,59 @@ cargo run --release --example compute_hashes -- --file ./canister_state.bin > ch
 ```
 
 The hash of each chunk is saved in `chunk_hashes.txt`.
+
+## 7. Hardcode the caches into the uploader canister
+
+```
+cp bootstrap/chunk_hashes.txt bootstrap/uploader/src/chunk_hashes.txt
+```
+
+Update the `state_size` in `bootstrap/uploader/src/main.rs` to be equal to the state size in 64k blocks, i.e. `echo $(( $(stat -c %s bootstrap/canister_state.bin) / 65536 ))`
+
+## 8. Deploy the uploader canister
+
+`dfx` version `0.12.1` is known to work
+
+Add desired network to the `dfx.json` and set `NETWORK` to match it.
+
+Create, build and install:
+
+```
+dfx canister --network=$NETWORK create uploader --no-wallet
+dfx build $uploader --network=$NETWORK
+dfx canister install uploader --network=$NETWORK
+```
+
+## 9. Upload the state
+
+Set `CANISTER` to match canister id, `IP` to match one of the nodes.
+
+```
+cd uploader
+cargo run --example upload -- --canister-id $CANISTER --state ../canister_state.bin --ic-network http://[${IP}]:8080  --fetch-root-key
+```
+
+If the desired node gets unhealthy, the uploader crashes but it continues to upload on restart. Takes few hours to few days.
+
+## 10. Hardcode the canister id into the replica binary
+
+Set the bitcoin canister id in `rs/config/src/execution_environment.rs` to match the uploader canister id. Note that "mainnet" and "testnet" stand for the BTC netwrok version, not IC. Build image and upgrade the subnet.
```

Comment:
That is no longer necessary. You can now deploy a testnet with the flag `--allow-specified-ids` and with that you can create the canister with the same ID as the bitcoin canister in production.

Reply:
Ack, I need to test how it works